api/v1alpha1/clusterorder_types.go

```diff
@@ -63,20 +66,31 @@ const (
 	// ClusterOrderConditionControlPlaneAvailable means the cluster control plane is ready
 	ClusterOrderConditionControlPlaneAvailable ClusterOrderConditionType = "ControlPlaneAvailable"
 
-	// ClusterOrderConditionNodePoolAvailable means the node pool has the correct number of nodes
-	ClusterOrderConditionNodePoolAvailable ClusterOrderConditionType = "NodePoolAvailable"
-
 	// ClusterOrderConditionAvailable means the cluster is available
 	ClusterOrderConditionAvailable ClusterOrderConditionType = "Available"
 )
 
+// ClusterOrderClusterReferenceType contains a reference to the namespace created by this ClusterOrder
+type ClusterOrderClusterReferenceType struct {
+	// Namespace that contains the HostedCluster resource
+	ClusterNamespace   string `json:"clusterNamespace"`
+	ClusterName        string `json:"clusterName"`
+	ServiceAccountName string `json:"serviceAccountName"`
+}
+
 // ClusterOrderStatus defines the observed state of ClusterOrder
 type ClusterOrderStatus struct {
 	// Phase provides a single-value overview of the state of the ClusterOrder
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Enum=Unknown;Accepted;Progressing;Failed;Ready
 	Phase ClusterOrderPhaseType `json:"phase,omitempty"`
 
 	// Conditions holds an array of metav1.Condition that describe the state of the ClusterOrder
 	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+
+	// Reference to the namespace that contains the HostedCluster resource
+	ClusterReference ClusterOrderClusterReferenceType `json:"clusterReference"`
```

Comment:
I think we want this whole field to be optional. It's a clean aggregation point; we either have a reference or we don't. It's common in k8s APIs to use a `nil` value to mean "we don't have any info on this sub-object".

IIRC making it optional just requires making this a pointer, and saying it's optional in the json metadata tag.